web_programming/fetch_anime_and_play.py

```diff
@@ -0,0 +1,188 @@
+from xml.dom import NotFoundErr
+
+import requests
+from bs4 import BeautifulSoup, NavigableString
+from fake_useragent import UserAgent
+
+BASE_URL = "https://ww1.gogoanime2.org"
+
+
+def search_scraper(anime_name: str) -> list:
+
+    """[summary]
+
+    Take an url and
+    return list of anime after scraping the site.
+
+    >>> type(search_scraper("demon_slayer"))
+    <class 'list'>
+
+    Args:
+        anime_name (str): [Name of anime]
+
+    Raises:
+        e: [Raises exception on failure]
+
+    Returns:
+        [list]: [List of animes]
+    """
+
+    # concat the name to form the search url.
+    search_url = f"{BASE_URL}/search/{anime_name}"
+
+    response = requests.get(
+        search_url, headers={"UserAgent": UserAgent().chrome}
+    )  # request the url.
```

Comment:
Don't wrap a line just for a comment.
```suggestion
    # request the url.
    response = requests.get(search_url, headers={"UserAgent": UserAgent().chrome})
```